CMakeLists.txt

```diff
@@ -40,12 +40,17 @@ set(CMAKE_CXX_FLAGS "${common_flags} ${CMAKE_CXX_FLAGS}")
 set(CMAKE_C_FLAGS "${common_flags} ${CMAKE_C_FLAGS}")
 set(CMAKE_CXX_STANDARD 17)
 
+if(WITH_TEST)
```

Comment:
It is not required as you can override it in the upper layer.

Reply:
Unfortunately, this is necessary because these changes will not be overridden. Using the main CMake file might be possible when running tests separately, meaning we would need a separate build. Therefore, these will be distinct changes for CMake and the build, different from the main configuration.

Reply:
We should find another solution. Let's discuss.